cpp/arduino/avr/pgmspace.h

```diff
@@ -50,46 +57,69 @@ out.each { |l| puts d(l) }
+inline void *memcpy_P(void *dest, const void *src, size_t n) { return memcpy(dest, src, n); }
+inline char *strcat_P(char *dest, const char *src) { return strcat(dest, src); }
+inline const char *strchr_P(const char *s, int val) { return strchr(s, val); }
+inline int strcmp_P(const char *s1, const char *s2) { return strcmp(s1, s2); }
+inline char *strcpy_P(char *dest, const char *src) { return strcpy(dest, src); }
+inline size_t strcspn_P(const char *s, const char *reject) { return strcspn(s, reject); }
+// strlcat and strlcpy are AVR-specific and not entirely trivial to reimplement using strncat it seems
+//inline size_t strlcat_P(char *dst, const char *src, size_t siz) { return strlcat(dst, src, siz); }
+//inline size_t strlcpy_P(char *dst, const char *src, size_t siz) { return strlcpy(dst, src, siz); }
+//inline size_t strlcat_PF(char *dst, uint_farptr_t src, size_t n) { return strlcat(dst, (const char*)src, n); }
+//inline size_t strlcpy_PF(char *dst, uint_farptr_t src, size_t siz) { return strlcpy(dst, (const char*)src, siz); }
+inline int strncmp_P(const char *s1, const char *s2, size_t n) { return strncmp(s1, s2, n); }
+inline char *strncat_P(char *dest, const char *src, size_t len) { return strncat(dest, src, len); }
+inline char *strncpy_P(char *dest, const char *src, size_t n) { return strncpy(dest, src, n); }
+inline char *strpbrk_P(const char *s, const char *accept) { return (char*)strpbrk(s, accept); }
+inline const char *strrchr_P(const char *s, int val) { return strrchr(s, val); }
+inline size_t strspn_P(const char *s, const char *accept) { return strspn(s, accept); }
+inline char *strstr_P(const char *s1, const char *s2) { return (char*)strstr(s1, s2); }
+inline char *strtok_P(char *s, const char * delim) { return strtok(s, delim); }
+inline size_t strlen_PF(uint_farptr_t s) { return strlen((char*)s); }
+inline void *memcpy_PF(void *dest, uint_farptr_t src, size_t n) { return memcpy(dest, (const char*)src, n); }
+inline char *strcpy_PF(char *dst, uint_farptr_t src) { return strcpy(dst, (const char*)src); }
+inline char *strncpy_PF(char *dst, uint_farptr_t src, size_t n) { return strncpy(dst, (const char*)src, n); }
+inline char *strcat_PF(char *dst, uint_farptr_t src) { return strcat(dst, (const char*)src); }
+inline char *strncat_PF(char *dst, uint_farptr_t src, size_t n) { return strncat(dst, (const char*)src, n); }
+inline int strcmp_PF(const char *s1, uint_farptr_t s2) { return strcmp(s1, (const char*)s2); }
+inline int strncmp_PF(const char *s1, uint_farptr_t s2, size_t n) { return strncmp(s1, (const char*)s2, n); }
+inline char *strstr_PF(const char *s1, uint_farptr_t s2) { return (char*)strstr(s1, (const char*)s2); }
+inline int memcmp_PF(const void *s1, uint_farptr_t s2, size_t len) { return memcmp(s1, (const char*)s2, len); }
+inline size_t strlen_P(const char *src) { return strlen(src); }
+
+// TODO: These functions cannot be found on the CYGWIN test build for
```

Comment:
Can you open an issue about this?  I'm not sure from this comment how to articulate the problem, and I don't want to lose it

Reply:
Done: https://github.com/Arduino-CI/arduino_ci/issues/204